### PR TITLE
Account selector refactor

### PR DESCRIPTION
**Pull request by danielSbastos**
Hey,

In this PR I change the manner in which the account is fetched. Now, instead of being in the reducer/dispatch, it is done via a useEffect hook.

The current selected account is now accessible to the parent component App, meaning that any of the children can have access to it if necessary.

### CODE REVIEW

**Review by laurogripa — CHANGES_REQUESTED** [review]
Let's fix this warnings

File: src/components/Navbar.tsx [hunk 1]
```diff
@@ -64,15 +73,24 @@ const NavbarGallery = () => (
   </Button>
 )
 
-const NavbarAccount = () => {
-  const [accountAddress, setAccountAddress] = useState('')
+const NavbarAccount = ({ setAccount, account }: any) => {
```
Comment (alfakini) [inline src/components/Navbar.tsx:76]: Aways name the components with it's "type" as the end word.

```suggestion
const AccountNavbar = ({ setAccount, account }: any) => {
```

File: src/components/AccountSelector.tsx [hunk 2]
```diff
@@ -1,29 +1,16 @@
...
-  })
-
-  const initialAddress = keyringOptions?.length > 0 ? keyringOptions[0].key : ''
-
-  useEffect(() => {
-    setAccountAddress(initialAddress)
-    setAccountSelected(initialAddress)
-  }, [setAccountAddress, initialAddress])
+const Main = ({ setAccountAddress, accounts, initialAddress }: Props) => {
+  const [accountSelected, setAccountSelected] = useState(initialAddress)
```
Comment (alfakini) [inline src/components/AccountSelector.tsx]:
```suggestion
  const [selectedAccount, setSelectedAccount] = useState(initialAddress)
```

File: src/components/AccountSelector.tsx [hunk 3]
```diff
@@ -1,29 +1,16 @@
...
-import React, { useState, useEffect } from 'react'
+import React, { useState } from 'react'
 import { Dropdown, DropdownButton } from 'react-bootstrap'
 import styled from 'styled-components'
 import { useSubstrate } from '../substrate'
 
 type Props = {
   setAccountAddress: (address: string) => void
+  accounts: { name: string | undefined; address: string }[]
+  initialAddress: string
```
Comment (alfakini) [inline src/components/AccountSelector.tsx]:
```suggestion
  activeAccountAddress: string
```

File: src/components/Navbar.tsx [hunk 4]
```diff
@@ -16,13 +17,17 @@ type NavbarType = {
   showSocialIcons: boolean
   showGalleryButton: boolean
   showAccount: boolean
+  setAccount: (account: string) => void
```
Comment (alfakini) [inline src/components/Navbar.tsx:20]:
```suggestion
  setActiveAccount: (account: string) => void
```

File: src/components/Navbar.tsx [hunk 5]
```diff
@@ -16,13 +17,17 @@ type NavbarType = {
...
+  account: string
 }
 
 const Navbar = ({
   showBrandIcon,
   showSocialIcons,
   showGalleryButton,
   showAccount,
+  setAccount,
```
Comment (alfakini) [inline src/components/Navbar.tsx:29]:
```suggestion
  setActiveAccount,
```

File: src/components/Navbar.tsx [hunk 6]
```diff
@@ -16,13 +17,17 @@ type NavbarType = {
...
+  account,
```
Comment (alfakini) [inline src/components/Navbar.tsx:30]:
```suggestion
  activeAccount,
```

File: src/components/Navbar.tsx [hunk 7]
```diff
@@ -31,7 +36,11 @@ const Navbar = ({
         <CenterNav>
           {showGalleryButton ? <NavbarGallery /> : <></>}
           {showSocialIcons ? <NavbarSocial /> : <></>}
-          {showAccount ? <NavbarAccount /> : <></>}
+          {showAccount ? (
+            <NavbarAccount setActiveAccount={setAccount} account={account} />
```
Comment (alfakini) [inline src/components/Navbar.tsx]:
```suggestion
            <NavbarAccount setActiveAccount={setActiveAccount} activeAccount={activeAccount} />
```

File: src/components/Navbar.tsx [hunk 8]
```diff
@@ -64,15 +73,32 @@ const NavbarGallery = () => (
   </Button>
 )
 
-const NavbarAccount = () => {
-  const [accountAddress, setAccountAddress] = useState('')
+const NavbarAccount = ({
+  setActiveAccount,
+  account,
```
Comment (alfakini) [inline src/components/Navbar.tsx:78]:
```suggestion
  activeAccount,
```

File: src/components/Navbar.tsx [hunk 9]
```diff
@@ -64,15 +73,32 @@ const NavbarGallery = () => (
...
+}: {
+  setActiveAccount: (account: string) => void
+  account: string
```
Comment (alfakini) [inline src/components/Navbar.tsx:81]:
```suggestion
  setActiveAccount: (activeAccount: string) => void
  activeAccount: string
```

File: src/components/Navbar.tsx [hunk 10]
```diff
@@ -64,15 +73,32 @@ const NavbarGallery = () => (
...
 
   return (
     <>
-      {accountAddress ? (
-        <AccountSelector setAccountAddress={setAccountAddress} />
+      {accounts.length != 0 && account ? (
+        <AccountSelector
+          accounts={accounts}
+          initialAddress={account}
+          setAccountAddress={setActiveAccount}
```
Comment (alfakini) [inline src/components/Navbar.tsx]:
```suggestion
          activeAccount={activeAccount}
          setActiveAccount={setActiveAccount}
```

File: src/helpers/fetchAccounts.ts [hunk 11]
```diff
@@ -0,0 +1,37 @@
+import { web3Accounts, web3Enable } from '@polkadot/extension-dapp'
+import keyring from '@polkadot/ui-keyring'
+import { config } from '../substrate/config'
+
+const fetchAccounts = (
+  setAccounts: (
+    accounts: { name: string | undefined; address: string }[],
+  ) => void,
+  setAccountAddress: (account: string) => void,
```
Comment (alfakini) [inline src/helpers/fetchAccounts.ts:9]:
```suggestion
  setActiveAccount: (activeAccount: string) => void,
```

File: src/helpers/fetchAccounts.ts [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+
+      keyring.loadAll({ ss58Format: 42, type: 'sr25519' })
+
+      const accounts = allAccounts.map((account) => ({
+        name: account.meta.name,
+        address: account.address,
+      }))
+
+      setAccounts(accounts)
+      setAccountAddress(accounts[0].address)
```
Comment (alfakini) [inline src/helpers/fetchAccounts.ts:28]:
```suggestion
      setActiveAccount(accounts[0].address)
```

File: src/pages/App.tsx [hunk 13]
```diff
@@ -8,6 +9,7 @@ import { Home } from './Home'
 
 function Main() {
   const { apiState, apiError } = useSubstrate()
+  const [account, setAccount] = useState<string>('')
```
Comment (alfakini) [inline src/pages/App.tsx:12]:
```suggestion
  const [activeAccount, setActiveAccount] = useState<string>('')
```

File: src/pages/App.tsx [hunk 14]
```diff
@@ -21,7 +23,7 @@ function Main() {
     <StyledMain fluid>
       <Switch>
         <Route exact path="/">
-          <Home />
+          <Home account={account} setAccount={setAccount} />
```
Comment (alfakini) [inline src/pages/App.tsx:26]:
```suggestion
          <Home activeAccount={activeAccount} setActiveAccount={setActiveAccount} />
```

File: src/pages/Home.tsx [hunk 15]
```diff
@@ -4,10 +4,21 @@ import { Navbar } from '../components/Navbar'
 import Canary from '../static/canary.svg'
 import KappaSigmaMuTitle from '../static/kappa-sigma-mu-title.svg'
 
-const Home = () => {
+const Home = ({
+  setAccount,
+  account,
```
Comment (alfakini) [inline src/pages/Home.tsx:9]:
```suggestion
  setActiveAccount,
  activeAccount,
```

File: src/pages/Home.tsx [hunk 16]
```diff
@@ -4,10 +4,21 @@ import { Navbar } from '../components/Navbar'
...
+}: {
+  setAccount: (account: string) => void
+  account: string
```
Comment (alfakini) [inline src/pages/Home.tsx:12]:
```suggestion
  setActiveAccount: (activeAccount: string) => void
  activeAccount: string
```

File: src/pages/Home.tsx [hunk 17]
```diff
@@ -4,10 +4,21 @@ import { Navbar } from '../components/Navbar'
...
+}): JSX.Element => {
   return (
     <>
-      <Navbar showSocialIcons showAccount />
+      <Navbar
+        showSocialIcons
+        showAccount
+        setAccount={setAccount}
+        account={account}
```
Comment (alfakini) [inline src/pages/Home.tsx:20]:
```suggestion
        setActiveAccount ={setActiveAccount}
        activeAccount ={activeAccount}
```

File: src/components/Navbar.tsx [hunk 18]
```diff
@@ -16,13 +17,17 @@ type NavbarType = {
   showSocialIcons: boolean
   showGalleryButton: boolean
   showAccount: boolean
+  setAccount: (account: string) => void
+  account: string
```
Comment (alfakini) [inline src/components/Navbar.tsx:21]:
```suggestion
  activeAccount: string
```

File: src/components/Navbar.tsx [hunk 19]
```diff
@@ -31,7 +36,11 @@ const Navbar = ({
         <CenterNav>
           {showGalleryButton ? <NavbarGallery /> : <></>}
           {showSocialIcons ? <NavbarSocial /> : <></>}
-          {showAccount ? <NavbarAccount /> : <></>}
+          {showAccount ? (
+            <AccountNavbar setActiveAccount={setAccount} account={account} />
```
Comment (alfakini) [inline src/components/Navbar.tsx:40]:
```suggestion
            <AccountNavbar setActiveAccount={setAccount} activeAccount={activeAccount} />
```

File: src/components/Navbar.tsx [hunk 20]
```diff
@@ -64,15 +73,32 @@ const NavbarGallery = () => (
...
 
   return (
     <>
-      {accountAddress ? (
-        <AccountSelector setAccountAddress={setAccountAddress} />
+      {accounts.length != 0 && account ? (
+        <AccountSelector
+          accounts={accounts}
+          activeAddress={account}
+          setAccountAddress={setActiveAccount}
```
Comment (alfakini) [inline src/components/Navbar.tsx:93]:
```suggestion
          activeAddress={activeAddress}
          setActiveAccount={setActiveAccount}
```

**Review by alfakini — APPROVED** [review]

**Review by murilooon — APPROVED** [review]

**Pull request merged**